source/opt/local_access_chain_convert_pass.cpp

```diff
@@ -247,9 +262,11 @@ Pass::Status LocalAccessChainConvertPass::ConvertLocalAccessChains(
     for (auto ii = bi->begin(); ii != bi->end(); ++ii) {
       switch (ii->opcode()) {
         case SpvOpLoad: {
+          if (HasOpenCL100DebugRef(&*ii)) break;
```

Comment:
I'm wondering if this restriction might be a bit too much.  If we do not change the access chain, then many other optimizations could be disabled later one, and they will be needed for legalization.  We should talk about which cases you are worried about, and how they can be handled.

For example, if the result of the load is used in a DebugValue, then you change the DebugValue to use the result of toe extract.  I believe that will already happen.

Reply:
I agree that this is too much. I just realized that the result of `ReplaceAccessChainLoad()` does not change the value -> the original one is a load of a component of a composite vs. the new one is an extraction of a composite.

I simply remove this if-statement.